runtime/src/main/java/org/corfudb/runtime/object/DiskBackedSMRSnapshot.java

```diff
@@ -122,16 +123,17 @@ public void release() {
     }
 
     public <K, V> RocksDbEntryIterator<K, V> newIterator(ISerializer serializer, Transaction transaction) {
-        // When newIterator is invoked, it's possible that this snapshot has since been invalidated.
+        // When getIterator is invoked, it's possible that this snapshot has since been invalidated.
         // Requesting an iterator from the RocksDB transaction with an invalid snapshot/readOptions causes
         // an internal assertion failure. Hence, CheckedRocksIterator performs necessary validation before
         // creating the new iterator.
-        return executeInSnapshot(readOptions -> {
-            RocksDbEntryIterator<K, V> iterator = new RocksDbEntryIterator<>(
-                    transaction.getIterator(readOptions),
-                    serializer, readOptions, lock, LOAD_VALUES);
-            set.add(iterator);
-            return iterator;
+        final RocksIterator rawIterator = executeInSnapshot(readOptions -> {
+            return transaction.getIterator(readOptions);
         });
+
+        final RocksDbEntryIterator<K, V> iterator = new RocksDbEntryIterator<>(
+                rawIterator, serializer, readOptions, lock, LOAD_VALUES);
+        set.add(iterator);
```

Comment:
Discussed offline, but to summarize, this can cause a resource leak when `release()` is invoked after the creation of the iterator but before it's added to the set.

Reply:
Fixes.